pkgs/test_reflective_loader/lib/test_reflective_loader.dart

```diff
@@ -154,12 +154,16 @@ void _addTestsIfTopLevelSuite() {
         if (allGroups || group.isSolo) {
           for (var test in group.tests) {
             if (allTests || test.isSolo) {
-              test_package.test(test.name, test.function,
+              test_package.test(test.name, () async {
+                await group.ensureSetUpClass();
+                test.function();
+              },
                   timeout: test.timeout,
                   skip: test.isSkipped,
                   location: test.location);
             }
           }
+          test_package.tearDownAll(() => group.tearDownClass());
```

Comment:
If we have two groups, first with `test_` methods, second with `test_` methods and one `solo_test_` method. I think we will enter both groups (because `allGroups` will be `true`), and will invoke `group.tearDownClass()` for the first group, without running any test in it, so no `group.setUpClass()` (because `allTests` is `false`).

I totally might be wrong here, you have your code change, so can run an experiment locally :-)